src/amt/cim.ts

```diff
@@ -3,56 +3,251 @@
-        response = this.wsmanMessageCreator.createXml(header, body)
-        break
-      case 'Pull':
-        header = this.wsmanMessageCreator.createHeader(CIM_Actions.PULL, this.resourceUriBase + 'CIM_ServiceAvailableToElement', messageId)
-        if (enumerationContext != null) { body = this.wsmanMessageCreator.createBody(CIM_Methods.PULL, enumerationContext) } else { body = null }
-        response = this.wsmanMessageCreator.createXml(header, body)
-        break
-      case 'Enumerate':
-        header = this.wsmanMessageCreator.createHeader(CIM_Actions.ENUMERATE, this.resourceUriBase + 'CIM_ServiceAvailableToElement', messageId)
-        body = this.wsmanMessageCreator.createBody(CIM_Methods.ENUMERATE)
-        response = this.wsmanMessageCreator.createXml(header, body)
-        break
-      case 'Release':
-        header = null // TODO: implement header
-        body = null // TODO: implement body
-        response = null // TODO: implement response
-        break
-      default:
-        response = null // TODO: Error Handling
+        return this.get(CIM_Actions.GET, CIM_Classes.CIM_COMPUTER_SYSTEM_PACKAGE, messageId)
+      case 'Enumerate':
+        return this.enumerate(CIM_Actions.ENUMERATE, CIM_Classes.CIM_COMPUTER_SYSTEM_PACKAGE, messageId)
+      default:
+        return null
+    }
+  }
+
+  cim_SystemPackaging = (method: CIM_Methods, messageId: String, enumerationContext?: String): String => {
+    switch (method) {
+      case 'Enumerate':
+        return this.enumerate(CIM_Actions.ENUMERATE, CIM_Classes.CIM_SYSTEM_PACKAGING, messageId)
+      case 'Pull':
+        if (enumerationContext != null) {
+          return this.pull(CIM_Actions.PULL, CIM_Classes.CIM_SYSTEM_PACKAGING, messageId, enumerationContext)
+        } else {
+          return null
+        }
+      default:
+        return null
+    }
+  }
+
+  cim_KVMRedirectionSAP = (method: CIM_Methods, messageId: String): String => {
+    switch (method) {
+      case 'Get':
+        return this.get(CIM_Actions.GET, CIM_Classes.CIM_KVM_REDIRECTION_SAP, messageId)
+      default:
+        return null
+    }
+  }
+
+  cim_Chassis = (method: CIM_Methods, messageId: String): String => {
+    switch (method) {
+      case 'Get':
+        return this.get(CIM_Actions.GET, CIM_Classes.CIM_CHASSIS, messageId)
+      default:
+        return null
+    }
+  }
+
+  cim_Chip = (method: CIM_Methods, messageId: String, enumerationContext?: String): String => {
+    switch (method) {
+      case 'Enumerate':
+        return this.enumerate(CIM_Actions.ENUMERATE, CIM_Classes.CIM_CHIP, messageId)
+      case 'Pull':
+        if (enumerationContext != null) {
+          return this.pull(CIM_Actions.PULL, CIM_Classes.CIM_CHIP, messageId, enumerationContext)
+        } else {
+          return null
+        }
+      default:
+        return null
+    }
+  }
+
+  cim_Card = (method: CIM_Methods, messageId: String): String => {
+    switch (method) {
```

Comment:
I know you're going for consistency with these, but with only one/two cases i'm not sure legibility is the greatest here -- are there more cases that are coming?

Reply:
All of these potentially have up to 6 methods available (Get, Pull, Enumerate, Put, Subscribe, Unsubscribe) that come from the base CIM model.  Currently I'm just implementing the ones that we use.